File: backend/app/services/matching.py
from app.services.llm import match_student_to_needs
from app.database import get_supabase


async def get_match_suggestions(application_id: str) -> list[dict]:
    supabase = get_supabase()

    app = supabase.table("credit_applications").select("*").eq("id", application_id).execute()
    if not app.data or not app.data[0].get("ai_score"):
        raise ValueError("Application must be scored first")
    app_data = app.data[0]

    student = supabase.table("student_profiles").select("*").eq("id", app_data["student_id"]).execute()

    needs = supabase.table("company_needs").select("*, company_profiles!inner(company_name, sector)").eq("status", "open").execute()

    result = await match_student_to_needs(
        student_data={**student.data, "justification": app_data["justification"]},
        score_data={
            "total_score": app_data["ai_score"],
            "breakdown": app_data["ai_score_breakdown"],
        },
        company_needs=needs.data or [],
    )

    return result

File: backend/src/services/matching.ts
import { supabase } from "../database";
import { matchStudentToNeeds } from "./llm";

export async function getMatchSuggestions(applicationId: string): Promise<Record<string, any>[]> {
  const app = await supabase
    .from("credit_applications")
    .select("*")
    .eq("id", applicationId);

  if (!app.data || app.data.length === 0 || !app.data[0].ai_score) {
    throw new Error("Application must be scored first");
  }
  const appData = app.data[0];

  const student = await supabase
    .from("student_profiles")
    .select("*")
    .eq("id", appData.student_id);

  const needs = await supabase
    .from("company_needs")
    .select("*, company_profiles!inner(company_name, sector)")
    .eq("status", "open");

  const result = await matchStudentToNeeds(
    { ...(student.data?.[0] || {}), justification: appData.justification },
    {
      total_score: appData.ai_score,
      breakdown: appData.ai_score_breakdown,
    },
    needs.data || [],
  );

  return result;
}
